src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+            if (invocationOperation.Arguments.Length < 1)
+            {
+                return;
+            }
+
+            // Check that the object is a StringBuilder
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+            {
+                return;
+            }
+
+            IMethodSymbol appendStringMethod = stringBuilderType
+                .GetMembers("Append")
+                .OfType<IMethodSymbol>()
+                .WhereAsArray(s =>
+                    s.Parameters.Length == 1 &&
+                    s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                .FirstOrDefault();
+            if (appendStringMethod is null)
+            {
+                return;
+            }
+
+            if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+            {
+                return;
+            }
+
+            ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+            IArgumentOperation firstArgument = arguments[0];
+
+            // We don't handle class fields. Only local declarations
+            if (!(firstArgument.Value is ILocalReferenceOperation argumentValue))
+            {
+                return;
+            }
+
+            ILocalSymbol localArgumentDeclaration = argumentValue.Local;
+
+            // If there are multiple declarations, bail
+            var semanticModel = context.Operation.SemanticModel;
+            var cancellationToken = context.CancellationToken;
+            SyntaxReference declaringSyntaxReference = localArgumentDeclaration.DeclaringSyntaxReferences.First();
+            var variableDeclarator = semanticModel.GetOperationWalkingUpParentChain(declaringSyntaxReference.GetSyntax(cancellationToken), cancellationToken);
+            if (variableDeclarator is IVariableDeclaratorOperation variableDeclaratorOperation)
+            {
+                IVariableDeclarationOperation variableDeclarationOperation = (IVariableDeclarationOperation)variableDeclaratorOperation.Parent;
+                if (variableDeclarationOperation == null)
+                {
+                    return;
+                }
+
+                IVariableDeclarationGroupOperation variableGroupDeclarationOperation = (IVariableDeclarationGroupOperation)variableDeclarationOperation.Parent;
+                if (variableGroupDeclarationOperation.Declarations.Length != 1)
+                {
+                    return;
+                }
+
+                if (variableDeclarationOperation.Declarators.Length != 1)
+                {
+                    return;
+                }
+            }
+
+            // Ok, single variable declaration
+            if (localArgumentDeclaration.HasConstantValue && localArgumentDeclaration.ConstantValue is string unitString && unitString.Length == 1)
```

Comment:
Is this rule not catching what I'd expect to be the most common case: `Append("s")`?  It's nice that the rule is able to handle declared consts, but in my experience, the far more common case to handle is when the literal passed in is just a single character but with double quotes instead of single quotes.

Reply:
Yeap, for some reason I decided not to tag that in the analyzer. Fixed now. Opened https://github.com/dotnet/runtime/pull/34843 too to fix all the cases in dotnet/runtime